### PR TITLE
Remove dependency on 'lastversion' package

### PR DESCRIPTION
**Pull request by jayaddison**
~~This changeset relies on hypothetical release asset filename updates proposed by https://github.com/grocy/grocy/pull/641~~

Remove the dependency on `lastversion`, and allow the user to specify a `grocy` [release version](https://github.com/grocy/grocy/releases) at build-time.

### CODE REVIEW

**Comment by berrnd** [comment]
See https://github.com/grocy/grocy/issues/640#issuecomment-602247220.

I would suggest to just use https://releases.grocy.info/latest.

(I'm not maintaining grocy/grocy-docker, /CC @talmai.)

**Comment by jayaddison** [comment]
Given what I now understand about the contents of the file from https://releases.grocy.info/latest, I'll begin using that as a data source for installation of a development environment.

For my use case it would still be beneficial to replace the `lastversion` step; something about this line is causing compatibility issues when I attempt to build an OCI container image using [buildah](https://buildah.io/).

Although the true problem may lie somewhere in the evaluation of the shell command, it would still allow me to continue past the problem (and removing dependencies without functionality loss always feels good anyway).

**Comment by talmai** [comment]
The whole reason we transition to lastversion, was to have a simple way to tag the different versions when building new images. Although, obviously, we never got that far. I'd be totally open to having less dependencies, but I'd hate to always grab latest without knowing what specific version it was. Any suggestion?

**Comment by jayaddison** [comment]
Good question :)

It seems like we have three options:

- https://api.github.com/repos/grocy/grocy/releases/latest - retrieve version information and a download URL from GitHub's API for the project
- https://releases.grocy.info/latest - request the latest file contents and extract a version from these
- https://pypi.org/project/lastversion - use `lastversion` to retrieve the release version and download URL

Verifying that the download really contains the same code published on GitHub is tricky, and my sense is that (1) and (3) would help us most for that, since we can lean on GitHub to provide verification.

(3) has the minor disadvantage that we have both `lastversion` and GitHub in our chain of trust to verify the file.  With (2) we trust @berrnd :)

**Comment by jayaddison** [comment]
One disadvantage to options (1) and (3) is that GitHub may potentially rate-limit us at build-time.  I've experienced this problem while testing image builds locally.

#39 should vastly reduce that disadvantage by enabling (optional) authenticated GitHub API requests at build-time.  Authenticated requests can be made much more frequently - 60 unauthenticated requests per hour vs 5000 requests per hour as per the [current GitHub API documentation](https://developer.github.com/v3/#rate-limiting).

**Comment by jayaddison** [comment]
@berrnd Do you have any plans to publish signed versions of `grocy`, or otherwise provide download integrity checks?

**Comment by berrnd** [comment]
Don't know if there is an "offical" way to sign ZIP files, but of course we could for example publish SHA265 checksums of the ZIPs.

Maybe as an text file in the GitHub release assets.
And on `https://releases.grocy.info/checksum?<VERSION>`

Would that work?

File: Dockerfile-grocy [hunk 1]
```diff
@@ -38,11 +38,9 @@ RUN     apk update && \
         docker-php-ext-install -j${NPROC} gd
 
 # Extract application release package
-RUN     wget --header "Authorization: ${GITHUB_API_TOKEN}" -t 3 -T 30 -nv "https://github.com/grocy/grocy/releases/latest/download/grocy.tar.gz" && \
-        tar xzf grocy.tar.gz && \
-        rm grocy.tar.gz && \
-        mv grocy/* . && \
-        rm -r grocy-* && \
+RUN     wget --header "Authorization: ${GITHUB_API_TOKEN}" -t 3 -T 30 -nv -O "grocy.zip" "https://releases.grocy.info/latest" && \
```
Comment (jayaddison) [inline Dockerfile-grocy]: :warning: This is/was a potential API key leak for anyone running from this commit :/

Comment (jayaddison) [reply Dockerfile-grocy]: cc @berrnd ; I've pushed to remove the offending commit, but there was briefly a change here which could have had people sending GitHub API tokens your way.  I'm not sure there's a ton we can safely do about that, and I think it's unlikely that anyone experienced this, but just a FYI.  I may delete this thread shortly to lessen the chance this code path & broken build script would accidentally appear again in future.

**Comment by jayaddison** [comment]
> Don't know if there is an "official" way to sign ZIP files, but of course we could for example publish SHA265 checksums of the ZIPs.
> 
> Maybe as an text file in the GitHub release assets.
> And on `https://releases.grocy.info/checksum?<VERSION>`
> 
> Would that work?

That would help gain confidence that downloads had not been accidentally corrupted or truncated in-transit, which does help a little.

One area where GitHub provides a lot of benefit in terms of integrity and trust is that they are independent from the `grocy` infrastructure and their business relies on providing accurate verification assurances about code commits.

In order for users to be confident that code downloaded from `releases.grocy.info` really is the 'official' published version, I believe we'd need some kind of independent chain-of-trust and signing process for the releases.

There's some documentation for Debian users about how to verify download integrity here: https://www.debian.org/CD/verify - it's possible we could re-use some ideas from them or similar projects.  Or alternatively, continue to use GitHub as a trusted platform (easier short-term, in my opinion).

File: Dockerfile-grocy [hunk 2]
```diff
@@ -1,6 +1,8 @@
 FROM php:7.2-fpm-alpine
 LABEL maintainer="Talmai Oliveira <to@talm.ai>"
 
+ARG GROCY_VERSION=v2.6.1
```
Comment (jayaddison) [inline Dockerfile-grocy]: @berrnd @talmai Hopefully this is a good all-round approach:

- The grocy version number is specified clearly within the repository contents
- It's easy to override at image-build-time for development & testing purposes
- It doesn't require using fixed file names in GitHub release assets

Wdyr?

**Comment by jayaddison** [comment]
I'm going to keep on rolling forward with this, since it will help out for resolution of #41 (performing a new Docker Hub release).  Any after-the-merge feedback is still welcome.

**Pull request merged**